x-pack/plugin/core/src/test/java/org/elasticsearch/snapshots/SourceOnlySnapshotShardTests.java

```diff
@@ -135,11 +135,23 @@ public void testIncrementalSnapshot() throws IOException {
             totalFileCount = copy.getTotalFileCount();
             assertEquals(copy.getStage(), IndexShardSnapshotStatus.Stage.DONE);
         }
+        try (Engine.IndexCommitRef snapshotRef = shard.acquireLastIndexCommit(true)) {
+            IndexShardSnapshotStatus indexShardSnapshotStatus = IndexShardSnapshotStatus.newInitializing(shardGeneration);
+            SnapshotId snapshotId = new SnapshotId("test_1", "test_1");
+            final PlainActionFuture<String> future = PlainActionFuture.newFuture();
+            runAsSnapshot(shard.getThreadPool(), () -> repository.snapshotShard(shard.store(), shard.mapperService(), snapshotId, indexId,
+                snapshotRef.getIndexCommit(), indexShardSnapshotStatus, true, Collections.emptyMap(), future));
+            shardGeneration = future.actionGet();
+            IndexShardSnapshotStatus.Copy copy = indexShardSnapshotStatus.asCopy();
+            assertEquals(0, copy.getIncrementalFileCount());
```

Comment:
Without this change, we're always uploading `1` file here because we keep recreating the `segments_N`